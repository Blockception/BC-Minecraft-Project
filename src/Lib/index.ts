/*	Auto generated	*/

export * from "./mcattributes";
export * from "./mcdefinitions";
export * from "./mcignore";
export * from "./project";