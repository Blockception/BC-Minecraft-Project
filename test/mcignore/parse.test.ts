import { TestFilesFolder } from "../utillity.test";
import * as path from "path";
import assert = require("assert");
import { MCIgnore } from "../../src/main";

const Text1 = `OutputFolder
Temp
Template/something
!BP/**/*.json`;

suite("MCIgnore", () => {
  test("parse1", () => {
    let Ignores = MCIgnore.parse(Text1);

    assert.strictEqual(Ignores.patterns.length, 4);
    assert(Ignores.patterns.includes("OutputFolder"));
    assert(Ignores.patterns.includes("Temp"));
    assert(Ignores.patterns.includes("Template/something"));
    assert(Ignores.patterns.includes("!BP/**/*.json"));
  });

  test("loadSync file1", () => {
    const filepath = path.join(TestFilesFolder, "mcignore", "file1.mcignore");

    let Ignores = MCIgnore.loadSync(filepath);

    assert.strictEqual(Ignores.patterns.length, 4);
    assert(Ignores.patterns.includes("OutputFolder"));
    assert(Ignores.patterns.includes("Temp"));
    assert(Ignores.patterns.includes("Template/something"));
    assert(Ignores.patterns.includes("!BP/**/*.json"));
  });

  test("load file2", (done) => {
    const filepath = path.join(TestFilesFolder, "mcignore", "file2.mcignore");

    MCIgnore.load(filepath).then((Ignores) => {
      assert.strictEqual(Ignores.patterns.length, 4);
      assert(Ignores.patterns.includes("OutputFolder"));
      assert(Ignores.patterns.includes("Temp"));
      assert(Ignores.patterns.includes("Template/something"));
      assert(Ignores.patterns.includes("!BP/**/*.json"));

      done();
    });
  });
});
