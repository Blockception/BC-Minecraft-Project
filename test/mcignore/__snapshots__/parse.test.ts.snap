// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MCIgnore load file2 1`] = `
{
  "patterns": [
    "OutputFolder",
    "Temp",
    "Template/something",
    "!BP/**/*.json",
  ],
}
`;

exports[`MCIgnore loadSync file1 1`] = `
{
  "patterns": [
    "OutputFolder",
    "Temp",
    "Template/something",
    "!BP/**/*.json",
  ],
}
`;

exports[`MCIgnore parse1 1`] = `
{
  "patterns": [
    "OutputFolder",
    "Temp",
    "Template/something",
    "!BP/**/*.json",
  ],
}
`;
